onnxruntime/core/common/path.cc

```diff
@@ -253,18 +253,16 @@ Path& Path::Append(const Path& other) {
   return *this;
 }
 
-Path& Path::Concat(const PathString& string) {
-  components_.back() += string;
-  return *this;
-}
-
-Path& Path::ConcatIndex(const int index) {
-#ifdef _WIN32
-  auto index_str = std::to_wstring(index);
-#else
-  auto index_str = std::to_string(index);
-#endif
-  components_.back() += index_str;
+Path& Path::Concat(const PathString& value) {
+  auto first_separator = std::find_if(value.begin(), value.end(), IsPreferredPathSeparator);
+  if (first_separator != value.end()){
+    ORT_THROW("Concat tries to concatenate a string with separator. String: ", ToMBString(value));
```

Comment:
the message could more strongly indicate what's wrong. e.g., "Cannot concatenate with a string containing a path separator."
also could use ORT_ENFORCE